cmd/root/user_agent_command_exec_id_test.go

```diff
@@ -2,25 +2,18 @@ package root
 
 import (
 	"context"
-	"regexp"
 	"testing"
 
+	"github.com/databricks/cli/libs/command"
 	"github.com/databricks/databricks-sdk-go/useragent"
-	"github.com/google/uuid"
 	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
 )
 
 func TestWithCommandExecIdInUserAgent(t *testing.T) {
-	ctx := withCommandExecIdInUserAgent(context.Background())
+	ctx := command.MockExecId(context.Background(), "some-exec-id")
+	ctx = withCommandExecIdInUserAgent(ctx)
 
-	// Check that the command exec ID is in the user agent string.
+	// Check that the command exec ID is set in the user agent string.
 	ua := useragent.FromContext(ctx)
-	re := regexp.MustCompile(`cmd-exec-id/([a-f0-9-]+)`)
-	matches := re.FindAllStringSubmatch(ua, -1)
-
-	// Assert that we have exactly one match and that it's a valid UUID.
-	require.Len(t, matches, 1)
-	_, err := uuid.Parse(matches[0][1])
-	assert.NoError(t, err)
+	assert.Contains(t, ua, "cmd-exec-id/some-exec-id")
```

Comment:
If the mock function is only included to make this test not use a regexp, it can be omitted IMO.